src/components/modebar/manage.js

```diff
@@ -138,7 +138,7 @@ function getButtonGroups(gd, buttonsToRemove, buttonsToAdd) {
         addGroup(['hoverClosestGl2d']);
     }
     else if(hasCartesian) {
-        addGroup(['hoverClosestCartesian', 'hoverCompareCartesian']);
+        addGroup(['toggleSpikelines', 'hoverClosestCartesian', 'hoverCompareCartesian']);
```

Comment:
My reasoning was to put it beside `hovermode: closest` because of how spikeline functionality is tied to that mode. (Needed for any spikelines, and using the toggle will force you into that mode)